### PR TITLE
Mh nine

### PR DESCRIPTION
**Pull request by Meng-Heng**

### CODE REVIEW

**Comment by jahorton** [comment]
Note:  this appears to include all of the commits from #772.

File: developer/16.0/reference/errors/index.md [hunk 1]
```diff
@@ -0,0 +1,897 @@
+---
+title: Error Codes
+---
+
+## Specifications
+
+<table class="standard-table">
+<colgroup>
+<col style="width: 50%" />
+<col style="width: 50%" />
```
Comment (jahorton) [inline developer/16.0/reference/errors/index.md:10]: I'm going to hold off on this one for a bit given all the HTML-table output that remains in the file; it's probably worth a double-check, or at least another set of eyes, before proceeding.

**Review by jahorton — COMMENTED** [review]
Missed a few code-format blocks that got dropped from links in prior rounds.

At this point, the main issues to resolve are:
- that one page that's got tons of HTML table formatting
- The broken links - it seems that `_nul.php` (for example) has special handling that `_nul.md` does not.

Lots of this in the broken-links log:

```
 Getting links from: http://localhost:8055/developer/current-version/guides/develop/tutorial/step-6
├─BROKEN─ http://localhost:8055/developer/language/reference/index (HTTP_404)
--
Getting links from: http://localhost:8055/developer/engine/web/14.0/reference/interface
├─BROKEN─ http://localhost:8055/developer/language/reference/nul (HTTP_404)
```

Note that the pages are _index.md, _nul.md (and were _index.php, _nul.php before).

**Comment by darcywong00** [comment]
> Note that the pages are _index.md, _nul.md (and were _index.php, _nul.php before).
There's special sauce in the  .htaccess

https://github.com/keymanapp/help.keyman.com/blob/staging/.htaccess#L186

You'll prob want to split out to redirect to _nul.md

**Comment by jahorton** [comment]
Thanks; that's the detail I was missing.

File: .htaccess [hunk 2]
```diff
@@ -183,7 +183,7 @@ RewriteRule "^keyboard/(?!redir.php)([^/?]+)/([^/?]+)(/)?$" "keyboard/redir.php?
 # special case because of specific filenames : these don't redirect properly otherwise
 
 # Rewrite nul -> _nul, index -> _index
-RewriteRule "^developer/language/reference/(nul|index)(\.php)?$" "developer/language/reference/_$1.php" [END]
+RewriteRule "^developer/language/reference/(nul|index)(\.php|\.md)?$" "developer/language/reference/_$1$2"
```
Comment (jahorton) [inline .htaccess:186]: Note:  removing the `[END]` was necessary - we need the .md rewrite rule to kick in later.

File: developer/16.0/reference/errors/index.md (hunk 1 above)
Comment (jahorton) [reply developer/16.0/reference/errors/index.md]: I've gone ahead and "de-converted" the page.  That seems the simplest way to move forward.

**Review by jahorton — APPROVED** [review]

**Comment by jahorton** [comment]
Alright, finally cleaned up & all broken-link issues have been resolved.  Merging.

**Pull request merged**